raft/rawnode_test.go

```diff
@@ -342,6 +342,7 @@ func TestRawNodeProposeAndConfChange(t *testing.T) {
 				if len(rd.Entries) > 0 {
 					t.Fatal("expected no more entries")
 				}
+				rawNode.Advance(rd)
```

Comment:
This got caused by the Ready-Ready without Advance fatal you introduced, right?

Reply:
Right. The fatal is actually gone in these later revisions, but I've kept the testing changes because they don't hurt.

Interestingly, with the way that we're remembering `stepsOnAdvance` on the `RawNode`, it actually is now safe to Ready-Ready without Advance. Or at least, I think it is, whereas it clearly wasn't before. Either way, the library still says it's not supported.